packages/wdio-junit-reporter/src/index.ts

```diff
@@ -69,7 +69,7 @@ class JunitReporter extends WDIOReporter {
         specFileName: string,
         suite: SuiteStats
     ) {
-        const featureName = this._prepareName(suite.title)
+        const featureName = this.options.testSuiteName ? this._prepareName(this.options.testSuiteName({ name: this.options.testSuiteName.name, activeFeatureName: this._activeFeatureName })): this._prepareName(suite.title)
```

Comment:
If someone provides a `testSuiteName` function we should expect that the name is valid - no need to call `_prepareName` and have some weird side effects if also `suiteNameFormat` is provided.

```suggestion
        const featureName = typeof this.options.testSuiteName === 'function'
            ? this.options.testSuiteName({
                name: this.options.testSuiteName.name,
                activeFeatureName: this._activeFeatureName
            })
            : this._prepareName(suite.title)
```